tests/Integration/Behaviour/Features/Context/Domain/StateFeatureContext.php

```diff
@@ -56,6 +66,71 @@ public function __construct()
         $this->defaultLangId = $configuration->get('PS_LANG_DEFAULT');
     }
 
+    /**
+     * @When I add new state :stateReference with following properties:
+     *
+     * @param string $stateReference
+     * @param TableNode $table
+     */
+    public function createState(string $stateReference, TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+
+        try {
+            /** @var StateId $stateId */
+            $stateId = $this->getCommandBus()->handle(new AddStateCommand(
+                (int) Country::getIdByName($this->defaultLangId, $data['country']),
+                Zone::getIdByName($data['zone']),
+                $data['name'],
+                $data['iso_code'],
+                PrimitiveUtils::castStringBooleanIntoBoolean($data['enabled'])
+            ));
+
+            SharedStorage::getStorage()->set($stateReference, new State($stateId->getValue()));
+        } catch (CannotAddStateException | StateConstraintException | StateException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @When I edit state :stateReference with following properties:
+     *
+     * @param string $stateReference
+     * @param TableNode $table
+     */
+    public function editState(string $stateReference, TableNode $table): void
+    {
+        /** @var State $state */
+        $state = SharedStorage::getStorage()->get($stateReference);
+
+        $command = new EditStateCommand($state->id);
+
+        $data = $table->getRowsHash();
+        if (isset($data['name'])) {
+            $command->setName((string) $data['name']);
+        }
+        if (isset($data['iso_code'])) {
+            $command->setIsoCode((string) $data['iso_code']);
+        }
+        if (isset($data['enabled'])) {
+            $command->setActive(PrimitiveUtils::castStringBooleanIntoBoolean($data['enabled']));
+        }
+        if (isset($data['country'])) {
+            $command->setCountryId((int) Country::getIdByName($this->defaultLangId, $data['country']));
+        }
+        if (isset($data['zone'])) {
+            $command->setZoneId((int) Zone::getIdByName($data['zone']));
+        }
+
+        try {
+            $this->getCommandBus()->handle($command);
+
+            SharedStorage::getStorage()->set($stateReference, new State($command->getStateId()->getValue()));
```

Comment:
You don't need to reset this value, it's already been set in the creation endpoint, and the ID can't change while updating the state

Reply:
Fixed